examples/step-67/doc/intro.dox

```diff
@@ -0,0 +1,680 @@
+@f[
+\Delta t = \frac{Cr}{p^{1.5}}\left(\frac{1}{\max\left[\frac{\|\mathbf{u}\|}{h_u} +
+           \frac{c}{h_c}\right]}\right).
+@f]
+The power $p^{1.5}$ used for the polynomial scaling is heuristic and
+represents the closest fit for polynomial degrees between 1 and 8, see
+e.g. @cite SchoederKormann2018. In the limit of higher degrees, $p>10$, a
+scaling of $p^2$ is more accurate, related to the inverse estimates typically
+used for interior penalty methods. Regarding the <i>effective</i> mesh sizes
+$h_u$ and $h_c$ used in the formula, we note that the convective transport is
+directional. Thus an appropriate scaling is to use the element length in the
+direction of the velocity $\mathbf u$. The code below derives this scaling
+from the inverse of the Jacobian from the reference to real cell, i.e., we
+approximate $\frac{\|\mathbf{u}\|}{h_u} \approx \|J^{-1} \mathbf
+u\|_{\infty}$. The acoustic waves, instead, are isotropic in character, which
+is why we use the smallest feature size, represented by the smallest singular
+value of $J$, for the acoustic scaling $h_c$. Finally, we need to add the
+convective and acoustic limits, as the Euler equations can transport
+information with speed $\|\mathbf{u}\|+c$.
+
+In this tutorial program, we use a specific variant of explicit Runge--Kutta
+methods, which in general use the following update procedure from the state
+$\mathbf{w}_h^{n}$ at time $t^n$ to the new time $t^{n+1}$ with $\Delta t =
+t^{n+1}-t^n$:
+@f[
+\begin{aligned}
+\mathbf{K}_1 &= \mathcal M^{-1} \mathcal L_h\left(t^n, \mathbf{w}_h^n\right),
+\\
+\mathbf{K}_2 &= \mathcal M^{-1} \mathcal L_h\left(t^n+c_2\Delta t,
+                       \mathbf{w}_h^n + a_{21} \Delta t \mathbf{K}_1\right),
+\\
+&\vdots \\
+\mathbf{K}_s &= \mathcal M^{-1} \mathcal L_h\left(t^n+c_s\Delta t,
+  \mathbf{w}_h^n + \sum_{j=1}^{s-1} a_{sj} \Delta t \mathbf{K}_j),
+\\
+\mathbf{w}_h^{n+1} = \mathbf{w}_h^n + \Delta t\left(b_1 \mathbf{K}_1 +
+b_2 \mathbf{K}_2 + \ldots + b_s \mathbf{K}_s\right).
+\end{aligned}
+@f]
+The vectors $\mathbf{K}_i$, $i=1,\ldots,s$, in an $s$-stage scheme are
+evaluations of the operator at some intermediate state and used to define the
+end-of-step value $\mathbf{w}_h^{n+1}$ via some linear combination. The scalar
+coefficients in this scheme, $c_i$, $a_{ij}$, and $b_j$, are defined such that
+certain conditions are satisfied for higher order schemes, the most basic one
+being $c_i = \sum_{j=1}^{i-1}a_{ij}$. The parameters are typically collected in
+the form of a so-called Butcher tableau as (for a five-stage scheme)
+@f[
+\begin{array}{c|ccccc}
+0 \\
+c_2 & a_{21} \\
+c_3 & a_{31} & a_{32} \\
+c_4 & a_{41} & a_{42} & a_{43} \\
+c_5 & a_{51} & a_{52} & a_{53} & a_{54} \\
+\hline
+& b_1 & b_2 & b_3 & b_4 & b_5
+\end{array}
+@f]
+
+In this tutorial program, we use a subset of explicit Runge--Kutta methods,
+so-called low-storage Runge--Kutta methods (LSRK). The motivation for
```

Comment:
out of curiosity, is storing a few extra vectors really a big deal? I mean, you're not going to use a 20-stage RK method :-)